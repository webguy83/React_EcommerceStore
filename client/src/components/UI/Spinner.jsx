import React from 'react';
import styled from 'styled-components/macro';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCompass } from '@fortawesome/free-solid-svg-icons';

// css
const SpinnerContainer = styled.div`
    position: absolute;
    left: calc(50% - 4.5rem);
    top: calc(50% - 4.5rem);
`
// jsx

const Spinner = () => {
    return <SpinnerContainer>
        <FontAwesomeIcon spin icon={faCompass} size="9x" />
    </SpinnerContainer>
};

export default Spinner;