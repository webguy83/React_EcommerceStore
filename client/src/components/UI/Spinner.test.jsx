import Spinner from "./Spinner";
import React from "react";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";

describe("default tests for loader", () => {
  afterEach(cleanup);

  it("should render Spinner without errors", () => {
    const { asFragment } = render(<Spinner />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should have the loader class attached to the div", () => {
    const { getByTestId } = render(<Spinner />);
    expect(getByTestId("loader").className).toBe("loader");
  });
});
