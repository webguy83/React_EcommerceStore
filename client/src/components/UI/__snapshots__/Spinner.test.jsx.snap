// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default tests for loader should render Spinner without errors 1`] = `
<DocumentFragment>
  .c0 {
  position: absolute;
  left: calc(50% - 4.5rem);
  top: calc(50% - 4.5rem);
}

.c0 .loader {
  border: 0.8rem solid #f3f3f3;
  border-top: 0.8rem solid var(--prim);
  border-radius: 50%;
  width: 6.4rem;
  height: 6.4rem;
  -webkit-animation: spin 2s linear infinite;
  animation: spin 2s linear infinite;
}

<div
    class="c0"
  >
    <div
      class="loader"
      data-testid="loader"
    />
  </div>
</DocumentFragment>
`;
